app/controllers/users_controller.rb

```diff
@@ -170,7 +170,15 @@ def changed_email
     if !params[:new_email].blank? && reauthenticate
       new_email = params[:new_email]
 
-      if new_email != params[:email_confirmation]
+      # Please note: This comparison is not technically correct. According to
+      # RFC 5321, the local part of an email address is case sensitive, while the
+      # domain is case insensitive. That said, all major email providers treat
+      # the local part as case insensitive, so it would probably cause more
+      # confusion if we did this correctly.
+      #
+      # Also, email addresses are validated on the client, and will only contain
+      # a limited subset of ASCII, so we don't need to do a unicode casefolding pass.
+      if new_email.downcase != params[:email_confirmation].downcase
         flash.now[:error] = ts("Email addresses don't match! Please retype and try again")
```

Comment:
One last thing I just noticed, can you add a period to the end of this error message? Otherwise this looks good.

Reply:
Certainly, good catch!